### ISSUE TITLE
What is the difference between this and libdns/dnsupdate

### ISSUE THREAD

**Issue by conblem** [libdns/rfc2136#3]
Hello everybody I stumbled upon this library and the libdns/dnsupdate library.
Both seem to be using RFC 2136 for DNS updates it is unclear to me what the distinction is between these libraries.
One thing I found is that only this library has caddy-dns integration, so maybe thats it?

Cheers

**Comment by mholt** [comment]
Good question, I don't think I realized they were the same. :sweat_smile: From the `dnsupdate` package:

> DNS AXFR is used to list DNS records. DNS UPDATE is used to append, set and delete records.

So maybe that's one difference? I don't know much about these RFCs though so maybe the package author(s) can chime in. /cc @emersion

**Comment by mholt** [comment]
I'm totally sorry for not noticing it either. I had always heard "rfc 2136" but never really thought of it as "DNS UPDATE" -- but of course it's the same thing :man_facepalming: 

Would you be interested in merging/consolidating the two repos? I could add both of you as maintainers to this one, for instance.

**Comment by rahilarious** [comment]
> Would you be interested in merging/consolidating the two repos? I could add both of you as maintainers to this one, for instance.

it'd be great to merge repos. @emersion ?

**Comment by mholt** [comment]
Note that libdns is going to be formalizing its APIs and semantics here in an upcoming 1.0 beta release that should make some things easier but also a little different. (I'll update this with a link to the PR when I'm back at my computer)

**Comment by emersion** [comment]
I'm guessing this is https://github.com/libdns/libdns/pull/153? Very nice to see the API improved with well-defined semantics and better type safety!

**Comment by mholt** [comment]
Yep! I've been wanting to do it for a while but it just takes some time to gain real world experience first.

**Comment by gucci-on-fleek** [comment]
@emersion 
 
> * rfc2136 doesn't populate the `ID` field of records.

That will be "fixed" in libdns/libdns#153 (which removes the `ID` field entirely).

> * rfc2136 uses `reflect`, which I find a bit fragile/surprising.

I've removed that in [my fork for `libdns` v1.0](https://github.com/gucci-on-fleek/libdns-rfc2136/tree/libdns-1.0). But to be fair, [I was the one who added `reflect` in the first place](https://github.com/libdns/rfc2136/pull/8/files#diff-4caa05d97eba477e577a7721860f68293d7371001a42747af73e647b581a5e9bR23-R25).
